docs/guides/10-integrating-react.md

```diff
@@ -0,0 +1,111 @@
+
+```jsx
+render() {
+  // We can pass some properties as attributes
+  <as-range-slider id="range" min-value={this.state.minValue} max-value={this.state.maxValue}></as-range-slider>
+}
+
+componentDidMount(){
+  // But we need to use javascript to set callbacks
+  const rangeSlider = document.querySelector('#range');
+  rangeSlider.addEventListener('change', this.onRangeChange.bind(this));
+}
+
+onRangeChange(event){
+  console.log(event.detail);
+}
+```
+
+
+## Option 2: Wrap the component into a react element
+
+With this option we are going to create a React component that wraps the original web-component exposing a simple and react-friendly API.
+
+
+> See the full example [here](https://github.com/CartoDB/airship-demos/tree/master/react/widget)
+
+
+First we create a new [React Component]() to represent our widget, in this case a `category-widget`. In the `render` function we will return
+the airship category widget referenced using a [ref](https://reactjs.org/docs/refs-and-the-dom.html).
+
+
+```js
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef();
+  }
+
+  render() {
+    <as-category-widget ref={this.ref} heading={this.props.heading} description={this.props.description}/>
+  }
+}
+```
+
+Then we will list all availiable properties using [React proptypes](https://reactjs.org/docs/typechecking-with-proptypes.html), this tells react
+which attributes to accept.
+
+```js
+export default class App extends Component {
+   static defaultProps = {
+    categories: [],
+    showHeader: true,
+    showClearButton: true,
+    useTotalPercentage: false,
+    visibleCategories: Infinity,
+  }
+}
+```
+
+Then we set up all the bindings in the `componentDidMount` function binding every React property to the corresponding in the web component. We also associate the events, adding a native listener that connects to `_onSelectedChanged` function.
+
+```js
+export default class App extends Component {
+  componentDidMount() {
+    const widget = ref.current;
+
+    widget.showHeader = this.props.showHeader;
+    widget.showClearButton = this.props.showClearButton;
+    widget.useTotalPercentage = this.props.useTotalPercentage;
+    widget.visibleCategories = this.props.visibleCategories;
+    widget.categories = this.props.categories;
+    widget.addEventListener('categoriesSelected', this._onSelectedChanged.bind(this));
+  }
+
+  _onSelectedChanged(event) {
+    const { onSelectedChanged } = this.props;
+    onSelectedChanged && onSelectedChanged(event);
+  }
+}
+```
+
+With this steps we created a React wrapper over the native element that allows to pass everything as an attribute
+
+```js
+ <CategoryWidget
+  heading="Business Volume"
+  description="Description"
+  categories={categories}
+  onSelectedChanged={this.onSelectedChanged}
+  showClearButton={this.showClearButton}
+```
```

Comment:
The category widget has also methods that can be hooked up with the wrapping React component. Talk about them also since we have props, events and methods.

Reply:
eing? I dont follow you

Reply:
There is a `getSelectedCategories ` and a `clearSelection ` method in the category widget.

I don't know how we can expose those methods in the React component but we should.

Reply:
https://shripadk.github.io/react/tips/expose-component-functions.html

It seems that it can be done just using a regular method. The parent should get a ref to the component but it's not our job telling how to use it from the parent, we just have to expose the method in the guide.